analyses/cryptojacking_analysis.md

```diff
@@ -0,0 +1,209 @@
+
+There are ways to prevent cryptojacking: 
+1. Use anti-mining browser extensions
+2. Use script blockers - Mozilla has one called NoScript.  
+Read [[1]](https://www.makeuseof.com/tag/what-is-cryptojacking/ ) and [[3]](https://www.csoonline.com/article/3253572/internet/what-is-cryptojacking-how-to-prevent-detect-and-recover-from-it.html) for more details.
+
+**The purpose of this notebook is to investigate the prevalence of cryptojacking by analysing script calls obtained from a web crawl of Alexa's top 10K sites in November 2017.**   
+Detection algorithm: for each script call, check if any of cryptojacking hosts are pattern matched to script names in script tags. 
+The list of potential cryptojacking hosts (212 sites total) was obtained from the [adblock-nocoin-list GitHub repo.](https://github.com/hoshsadiq/adblock-nocoin-list/blob/master/hosts.txt)  
+Aspects of analysis:
+1. How many script calls were detected with cryptojacking?
+2. How many unique domains used cryptojacking scripts?
+3. Which domains were the "worst" in terms of number of scripts used?
+4. How many of the 212 cryptojacking hosts were used? 
+5. Which cryptojacking scripts were used the most? 
+   
+The findings will potentially inspire future research and analysis on cryptojacking usage, and influence blocking lists implemented by browsers such as Mozilla Firefox.
+
+Sources:  
+[[1]](https://www.makeuseof.com/tag/what-is-cryptojacking/ ) https://www.makeuseof.com/tag/what-is-cryptojacking/  
+[[2]](https://gwillem.gitlab.io/2017/11/07/cryptojacking-found-on-2496-stores/) https://gwillem.gitlab.io/2017/11/07/cryptojacking-found-on-2496-stores/  
+[[3]](https://www.csoonline.com/article/3253572/internet/what-is-cryptojacking-how-to-prevent-detect-and-recover-from-it.html) https://www.csoonline.com/article/3253572/internet/what-is-cryptojacking-how-to-prevent-detect-and-recover-from-it.html  
+[[4]](https://www.wired.com/story/cryptojacking-cryptocurrency-mining-browser/) https://www.wired.com/story/cryptojacking-cryptocurrency-mining-browser/  
+[[5]](https://www.theregister.co.uk/2018/02/27/ethical_coinhive/) https://www.theregister.co.uk/2018/02/27/ethical_coinhive/
+
+### Results: Summary
+
+Less than 0.02% of script calls are detected with cryptojacking.  
+However, it is important to note that cryptojacking code can be executed in order ways than by including the host .js script in a script tag. It can be disguised, stealthily executed in iframes,  or directly used in a function of a first party script. Users may also face redirect loops that eventually lead to a page with a mining script. Another reason for the low detection rate could be that the prominence/popularity of the analysed sites from the web crawl is enough to dissuade site owners from implementing obvious cryptojacking script usage.  
```

Comment:
"other ways" ?

Reply:
I wrote in the next line: "It can be disguised, stealthily executed in iframes,  or directly used in a function of a first party script. Users may also face redirect loops that eventually lead to a page with a mining script."

I read these methods in various articles, and I'm assuming they don't involve bluntly including "coinhive.js" in script tags. 

Please let me know if that's incorrect!

Reply:
Oh, sorry I meant that the text says "order ways" but it should say "other ways" ? just a typeo.

Reply:
ahhh didn't even notice, thanks!!